### PR TITLE
Adding fedora 31 bootenv

### PR DESCRIPTION
**Pull request by michaelrice**
This chgange adds support for Fedora 31 x86_64

This change also introduces a new param called kickstart-base-packages

Fixes: rackn/provision-server#182

Signed-off-by: Michael Rice <michael@michaelrice.org>

### CODE REVIEW

File: content/stages/fedora-31.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+Name: "fedora-31"
```
Comment (galthaus) [inline content/stages/fedora-31.yml]: We usually have the stage match the rest that have -install on the end.

File: task-library/workflows/fedora-base.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+Documentation: |
+  This workflow includes the DRP Runner in Fedora provisioning process for DRP.
+
+  After the install completes, the workflow installs the runner
+  in a waiting state so that DRP will automatically detect and start a
+  new workflow if the Machine.Workflow is updated.
+
+  NOTE: To enable, upload the Fedora ISO as per the fedora-31 BootEnv
+Stages:
+  - fedora-31
```
Comment (galthaus) [inline task-library/workflows/fedora-base.yaml]: switch to -install if you make the change above.

**Comment by galthaus** [comment]
Otherwise this looks good.  You may want to add a protection around the kexec reboot in the kickstart if we don't get that working.

**Comment by michaelrice** [comment]
> Otherwise this looks good. You may want to add a protection around the kexec reboot in the kickstart if we don't get that working.

https://docs.fedoraproject.org/en-US/fedora/rawhide/install-guide/appendixes/Kickstart_Syntax_Reference/  I looked and according to the docs what we are doing is correct and supported. There may be a bug upstream though. Ill check bugzilla

**Comment by michaelrice** [comment]
This could be related: https://bugzilla.redhat.com/show_bug.cgi?id=1672405  I have no way to test this from my vms because I cant switch consoles, drp logs dont show any errors, but at this point of the install process may have no way to capture anything because it should be rebooting..

**Pull request merged**